java/connector-node/risingwave-connector-service/src/main/java/com/risingwave/sourcenode/SourceRequestHandler.java

```diff
@@ -97,20 +107,74 @@ private void validateDbProperties(
         }
-        try (var conn =
-                DriverManager.getConnection(
-                        jdbcUrl,
-                        props.get(DbzConnectorConfig.USER),
-                        props.get(DbzConnectorConfig.PASSWORD))) {
-            // usernamed and password are correct
-            var dbMeta = conn.getMetaData();
 
-            LOG.debug("source schema: {}", validate.getTableSchema().getColumnsList());
-            LOG.debug("source pk: {}", validate.getTableSchema().getPkIndicesList());
+        String dbUser = props.get(DbzConnectorConfig.USER);
+        String dbPassword = props.get(DbzConnectorConfig.PASSWORD);
+        switch (validate.getSourceType()) {
+            case POSTGRES:
+                try (var validator =
+                        new PostgresValidator(
+                                jdbcUrl,
+                                dbUser,
+                                dbPassword,
+                                props,
+                                sqlStmts,
+                                validate.getTableSchema())) {
+                    validator.validateAll();
+                }
+                break;
+
+            case CITUS:
+                try (var coordinatorValidator =
+                        new PostgresValidator(
+                                jdbcUrl,
+                                dbUser,
+                                dbPassword,
+                                props,
+                                sqlStmts,
+                                validate.getTableSchema())) {
+                    coordinatorValidator.validateDistributedTable();
+                    coordinatorValidator.validateTableSchema();
+                }
+
+                var servers = props.get(DbzConnectorConfig.DB_SERVERS);
+                var workerAddrs = StringUtils.split(servers, ',');
+                var jdbcPrefix = getJdbcPrefix(validate.getSourceType());
+                for (String workerAddr : workerAddrs) {
+                    String workerJdbcUrl =
+                            jdbcPrefix
+                                    + "://"
+                                    + workerAddr
+                                    + "/"
+                                    + props.get(DbzConnectorConfig.DB_NAME);
+
+                    LOG.info("workerJdbcUrl {}", workerJdbcUrl);
+                    try (var workerValidator =
+                            new PostgresValidator(
+                                    workerJdbcUrl,
+                                    dbUser,
+                                    dbPassword,
+                                    props,
+                                    sqlStmts,
+                                    validate.getTableSchema())) {
+                        workerValidator.validateLogConfig();
+                        workerValidator.validatePrivileges();
+                    }
+                }
+
+                break;
+            case MYSQL:
```

Comment:
Just curious: why MySQL doesn't need a "Validator"?

Reply:
It will but not that urgent. Just because I want to implement the feature ASAP. I will refactor the validation part when I have bandwidth.